### DISCUSSION TITLE
Bug Report:

### DISCUSSION THREAD

**Discussion by A1foostbit** [KadeArchive/Kade-Engine#531] [Q&A]
so I was building the kade engine game and I got this source/HealthIcon.hx:40: characters 56-57 : Expected , or ) I tried everything and nothing worked pleas help

**Comment by shaderw0lf** [comment]
can we see your code..?

**Comment by A1foostbit** [comment]
what code

**Comment by A1foostbit** [comment]
package;

import flixel.FlxSprite;

class HealthIcon extends FlxSprite
{
	/**
	 * Used for FreeplayState! If you use it elsewhere, prob gonna annoying
	 */
	public var sprTracker:FlxSprite;

	public function new(char:String = 'bf', isPlayer:Bool = false)
	{
		super();
		
		loadGraphic(Paths.image('iconGrid'), true, 150, 150);

		antialiasing = true;
		animation.add('bf', [0, 1], 0, false, isPlayer);
		animation.add('bf-car', [0, 1], 0, false, isPlayer);
		animation.add('bf-christmas', [0, 1], 0, false, isPlayer);
		animation.add('bf-pixel', [21, 21], 0, false, isPlayer);
		animation.add('spooky', [2, 3], 0, false, isPlayer);
		animation.add('pico', [4, 5], 0, false, isPlayer);
		animation.add('mom', [6, 7], 0, false, isPlayer);
		animation.add('mom-car', [6, 7], 0, false, isPlayer);
		animation.add('tankman', [8, 9], 0, false, isPlayer);
		animation.add('face', [10, 11], 0, false, isPlayer);
		animation.add('dad', [12, 13], 0, false, isPlayer);
		animation.add('senpai', [22, 22], 0, false, isPlayer);
		animation.add('senpai-angry', [22, 22], 0, false, isPlayer);
		animation.add('spirit', [23, 23], 0, false, isPlayer);
		animation.add('bf-old', [14, 15], 0, false, isPlayer);
		animation.add('gf', [16], 0, false, isPlayer);
		animation.add('gf-christmas', [16], 0, false, isPlayer);
		animation.add('gf-pixel', [16], 0, false, isPlayer);
		animation.add('parents-christmas', [17, 18], 0, false, isPlayer);
		animation.add('monster', [19, 20], 0, false, isPlayer);
		animation.add('monster-christmas', [19, 20], 0, false, isPlayer);
		animation.add('beatbot', [24, 25], 0, false, isPlayer};
		animation.add('madbot', [26, 27], 0, false, isPlayer};
		animation.add('killbot', [28, 29], 0, false, isPlayer};
		animation.play(char);
		switch(char)
		{
			case 'bf-pixel' | 'senpai' | 'senpai-angry' | 'spirit' | 'gf-pixel':
				antialiasing = false;
		}

		scrollFactor.set();
	}

	override function update(elapsed:Float)
	{
		super.update(elapsed);

		if (sprTracker != null)
			setPosition(sprTracker.x + sprTracker.width + 10, sprTracker.y - 30);
	}
}



there you go

**Comment by ghost** [comment]
After seeing your code, here's the fix:
```haxe
	animation.add('beatbot', [24, 25], 0, false, isPlayer};
	animation.add('madbot', [26, 27], 0, false, isPlayer};
	animation.add('killbot', [28, 29], 0, false, isPlayer};
```
needs to be
```haxe
	animation.add('beatbot', [24, 25], 0, false, isPlayer);
	animation.add('madbot', [26, 27], 0, false, isPlayer);
	animation.add('killbot', [28, 29], 0, false, isPlayer);
```
